src/snowflake/telemetry/_internal/exporter/otlp/proto/log_exporter/__init__.py

```diff
@@ -0,0 +1,170 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+import abc
+import enum
+import logging
+import logging.config
+import threading
+import typing
+import opentelemetry.sdk.util.instrumentation as otel_instrumentation
+
+from opentelemetry.sdk import resources
+from opentelemetry.sdk._logs import export
+from opentelemetry.sdk import _logs
+from opentelemetry.util import types
+from snowflake.telemetry._internal.encoder.otlp.proto.common.log_encoder import (
+    serialize_logs_data
+)
+
+LOGGER_NAME_TEMP_ATTRIBUTE = "__snow.logging.temp.logger_name"
+FILEPATH_ATTRIBUTE = "code.filepath"
+FUNCTION_NAME_ATTRIBUTE = "code.function"
+
+
+class LogWriterResult(enum.Enum):
+    SUCCESS = 0
+    FAILURE = 1
+
+
+# LogWriter abstract class with one abstract method that can be overwritten:
+class LogWriter(abc.ABC):
+
+    @abc.abstractmethod
+    def write_logs(self, serialized_logs: bytes) -> LogWriterResult:
+        """quick"""
+
+
+class _ProtoLogExporter(export.LogExporter):
+
+    def __init__(self, log_writer: LogWriter):
+        self.log_writer = log_writer
+
+    def export(self, batch: typing.Sequence[_logs.LogData]):
+        result = self.log_writer.write_logs(serialize_logs_data(batch))
+        if result == LogWriterResult.FAILURE:
```

Comment:
noob question: Can we make the `LogWriterResult` to extend the `LogExportResult` and declare the equivalent values?

Reply:
It's such a basic enum, and LogExportResult is from an experimental module. I think it's better to just declare the two basic values independently.